pages/onboarding/index.tsx

```diff
@@ -1,15 +1,129 @@
-import { Box, Button, Container, Typography } from '@mui/material';
+import { Box, Button, Container, TextField, Typography } from '@mui/material';
+import axios from 'axios';
+import jwt from 'jsonwebtoken';
 import { NextPage } from 'next';
 import Link from 'next/link';
+import { useRouter } from 'next/router';
+import React, { useState } from 'react';
+import Cookies from 'universal-cookie';
+
+import createAxiosInstance from 'client/lib/axios';
+import { UserProfile } from 'server/types/User';
+import { AUTH_COOKIE_NAME } from 'shared/constants';
+
+interface FormData {
+  bio: string;
+  twitter_username: string;
+  linkedin_url: string;
+  github_username: string;
+  website: string;
+}
+
+interface Props {
+  user: UserProfile;
+}
+
+const OnboardingPage: NextPage<Props> = ({ user }: Props) => {
+  const [formData, setFormData] = useState<FormData>({
+    bio: user.bio,
+    twitter_username: user.twitter_username,
+    linkedin_url: user.linkedin_url,
+    github_username: user.github_username,
+    website: user.website,
+  });
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isFormFilled, setIsFormFilled] = useState<boolean>(false);
+
+  const router = useRouter();
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    setFormData(prevFormData => ({ ...prevFormData, [name]: value }));
+    setIsFormFilled(true);
```

Comment:
if they type something in, then delete it, this will remain true.  `isFormFilled` can instead just be a map over all the values of the `formData` object:

```ts
const isFormFilled = Object.values(formData).every((value) => !!value);
```

Reply:
This still doesn't solve the issue because the state is queued so its always on a 1 character delay with this setup.

example, I have a empty form and write "test" in bio then backspace it

based on this code 
```ts
  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = event.target;
    setFormData(prevFormData => ({ ...prevFormData, [name]: value }));
    console.log(formData);
    const isFormFilled = Object.values(formData).some(value => !!value);
    console.log(isFormFilled);
    setIsFormFilled(isFormFilled);
  };
```
```
index.tsx:55 {bio: '', twitter_username: '', linkedin_url: '', github_username: '', website: ''}
index.tsx:57 false
index.tsx:55 {bio: 't', twitter_username: '', linkedin_url: '', github_username: '', website: ''}
index.tsx:57 true
index.tsx:55 {bio: 'te', twitter_username: '', linkedin_url: '', github_username: '', website: ''}
index.tsx:57 true
index.tsx:55 {bio: 'tes', twitter_username: '', linkedin_url: '', github_username: '', website: ''}
index.tsx:57 true
index.tsx:55 {bio: 'test', twitter_username: '', linkedin_url: '', github_username: '', website: ''}
index.tsx:57 true
index.tsx:55 {bio: 'tes', twitter_username: '', linkedin_url: '', github_username: '', website: ''}
index.tsx:57 true
index.tsx:55 {bio: 'te', twitter_username: '', linkedin_url: '', github_username: '', website: ''}
index.tsx:57 true
index.tsx:55 {bio: 't', twitter_username: '', linkedin_url: '', github_username: '', website: ''}
index.tsx:57 true
```

at the end since there is a 1 character delay, I can still submit

![image](https://user-images.githubusercontent.com/68509315/236876273-0e234198-8a51-4342-848f-ac4e8afa2bc7.png)